sql/2020/20_Caching/resource_age_party_and_type_wise_groups.sql

```diff
@@ -0,0 +1,45 @@
+#standardSQL
+# Age of resources party, type wise in groups.
+CREATE TEMPORARY FUNCTION toTimestamp(date_string STRING)
+RETURNS INT64 LANGUAGE js AS '''
+  try {
+    var timestamp = Math.round(new Date(date_string).getTime() / 1000);
+    return isNaN(timestamp) || timestamp < 0 ? null : timestamp;
+  } catch (e) {
+    return null;
+  }
+''';
+
+SELECT
+  client,
+  party,
+  resource_type,
+  COUNTIF(age_weeks IS NOT NULL) AS requests_total,
+  COUNTIF(age_weeks < 0) AS age_neg,
+  COUNTIF(age_weeks = 0) AS age_0wk,
+  COUNTIF(age_weeks >= 1 AND age_weeks <= 7) AS age_1_to_7wk,
+  COUNTIF(age_weeks >= 8 AND age_weeks <= 52) AS age_8_to_52wk,
+  COUNTIF(age_weeks >= 53 AND age_weeks <= 104) AS age_gt_1y,
+  COUNTIF(age_weeks >= 105) AS age_gt_2y

```

Comment:
Should we do these as percentage to allow easier year on year comparison?

```suggestion
  SAFE_DIVIDE(COUNTIF(age_weeks < 0), COUNTIF(age_weeks IS NOT NULL)) AS age_neg_pct,
  SAFE_DIVIDE(COUNTIF(age_weeks = 0), COUNTIF(age_weeks IS NOT NULL)) AS age_0wk_pct,
  SAFE_DIVIDE(COUNTIF(age_weeks >= 1 AND age_weeks <= 7), COUNTIF(age_weeks IS NOT NULL)) AS age_1_to_7wk_pct,
  SAFE_DIVIDE(COUNTIF(age_weeks >= 8 AND age_weeks <= 52), COUNTIF(age_weeks IS NOT NULL)) AS age_8_to_52wk_pct,
  SAFE_DIVIDE(COUNTIF(age_weeks >= 53 AND age_weeks <= 104), COUNTIF(age_weeks IS NOT NULL)) AS age_gt_1y_pct,
  SAFE_DIVIDE(COUNTIF(age_weeks >= 105), COUNTIF(age_weeks IS NOT NULL)) AS age_gt_2y_pct
```

Reply:
Included review feedback. Kept both count, pct  (as graphs used a 100% stacked bar chart and calculated its own pct automatically). Instead of requests_total, renamed to requests_with_age.